test/events/component-view.test.js

```diff
@@ -0,0 +1,140 @@
+/*global require, describe, it, before, after, sinon, document */
+require('../setup');
+const assert = require("assert");
+const settings = require("../../src/javascript/core/settings");
+const send = require("../../src/javascript/core/send");
+const core = require("../../src/javascript/core");
+const session = require("../../src/javascript/core/session");
+const componentView = require("../../src/javascript/events/component-view");
+
+const config = {
+	context: {
+		product: 'desktop'
+	},
+	user: {
+		user_id: '123456'
+	}
+};
+
+function viewed (target) {
+	const event = new Event('intersect');
+	target.dispatchEvent(event);
+}
+
+function setMockIntersectionObserver (observeSpy, unobserveSpy) {
+	function mockIntersectionObserver(callback, options) {
+		this.callback = callback;
+		this.options = options;
+	}
+
+	mockIntersectionObserver.prototype.observe = function (target) {
+		target.addEventListener('intersect', () => {
+			this.callback([{ isIntersecting: true, target }]);
+		});
+		observeSpy(target);
+	};
+
+	mockIntersectionObserver.prototype.unobserve = function (target) {
+		unobserveSpy(target);
+	};
+
+	window.IntersectionObserver = mockIntersectionObserver;
+}
+
+
+describe('component:view', () => {
+
+	let targetComponent;
+	const observeSpy = sinon.spy();
+	const unobserveSpy = sinon.spy();
+
+	beforeEach(() => {
+		session.init();
+		send.init();
+		settings.set("config", config);
+		setMockIntersectionObserver(observeSpy, unobserveSpy);
+		sinon.spy(core, 'track');
+	});
+
+	afterEach(() => {
+		observeSpy.resetHistory();
+		unobserveSpy.resetHistory();
+		document.body.removeChild(targetComponent);
+		targetComponent = undefined;
+		core.track.restore();
+	});
+
+	context('with default props', () => {
+		beforeEach(() => {
+			targetComponent = document.createElement('div');
+			targetComponent.setAttribute('data-o-tracking-view', true);
+			targetComponent.innerHTML = 'Default component view elements';
+			document.body.appendChild(targetComponent);
+
+			componentView.init();
+			viewed(targetComponent);
+		});
+
+		it('should track an event for a component view ', () => {
```

Comment:
Would it be worth adding a case for `should only track an event for a component view once` ?

Reply:
@robsquires I thought That is a IntersectionObserver side test. It might be a bit tricky to test that?
It means `mockIntersectionObserver` has to have the code for `unobserve`.

There is a line to check `unobserve` is called. Is it not enough?
`assert.equal(unobserveSpy.calledOnce, true, "IntersectionObserver unobserved target");`

Reply:
Could you fire `viewed(targetComponent);` twice?